packages/dds/tree/src/simple-tree/arrayNode.ts

```diff
@@ -45,6 +47,19 @@ import {
 import { TreeNodeValid, type MostDerivedData } from "./treeNodeValid.js";
 import { getUnhydratedContext } from "./createContext.js";
 
+/**
+ * A covariant base type for {@link (TreeArrayNode:interface)}.
+ *
+ * This provides the readonly subset of TreeArrayNode functionality, and is used as the source interface for moves since that needs to be covariant.
+ * @privateRemarks
+ * Ideally this would just include `TreeNode, WithType<string, NodeKind.Array>` in the extends list but https://github.com/microsoft/TypeScript/issues/16936 prevents that from compiling.
+ * As a workaround around for this TypeScript limitation, the conflicting type intersection is wrapped in `Awaited` (which has no effect on the type in this case) which allows it to compile.
+ * @system @sealed @public
```

Comment:
This is system since we will likely want to reactor how we related readonly and non readonly versions of nodes when fixing the non-exact schema cases for the editing API surface to be contravariant. As this is used by a public API we can't make it alpha instead, but we might provide an alpha wrapper around it (that is non-system) at some point.